### DISCUSSION TITLE
Curseforge files "modpack page URL or slug identifier is required"

### DISCUSSION THREAD

**Discussion by linthalo77** [itzg/docker-minecraft-server#3114] [General]
Trying to set up my forge server but I'm unable to install mods on it. The server is running flawlessly but my curseforge files aren't getting installed because of the error:

` [init] Running as uid=1000 gid=1000 with /data as 'drwxr-xr-x 9 1000 1000 4096 Oct 19 18:54 /data'`
 `A modpack page URL or slug identifier is required`
 ` [init] ERROR failed to auto-install CurseForge modpack`

I have tried URL's instead of the slugs but it's not working. Tried comma as a seperator, same problem. The java version, minecraft version and mod platform should all match. Also tried Forge as type instead of auto_curseforge. 

Any ideas?

docker-compose: https://github.com/user-attachments/assets/2c8682f3-01a8-4c2b-bc66-d35cc8862b6b

**Comment by itzg** [comment]
You seem to be mixing up features. For modpacks use one of the variables here

https://docker-minecraft-server.readthedocs.io/en/latest/types-and-platforms/mod-platforms/auto-curseforge/#usage

**Comment by linthalo77** [comment]
Yeah I'm a bit confused on the terminology. 

My understanding is that a modpack is a collection of many mods (e. g. all-the-mods) and everything else is a just a "mod". 
Is that correct or is modpack meant as one single mod? If it's a single mod, how do I add multiple ones with CF_PAGE_URL? 
My aim is to add multiple individual mods not a big modpack.

**Reply by itzg** [reply]
A mod is a single mod and a modpack is a pack of mods. Modpacks also declare the modloader (Forge, Fabric, etc) and Minecraft version. 

If you want to install your own set of mods, [pick a modloader and set TYPE](https://docker-minecraft-server.readthedocs.io/en/latest/types-and-platforms/), set VERSION appropriately, and continue on with https://docker-minecraft-server.readthedocs.io/en/latest/mods-and-plugins/curseforge-files/

**Reply by linthalo77** [reply]
But that's exactly what I did? My three selected mods are all for Minecraft Version 1.20.1, have Forge as their mod loader and the mod platform is CurseForge. I set the files manually so it downloads the correct one e.g.
"https://www.curseforge.com/minecraft/mc-mods/chisels-bits/files/5203366"

My TYPE is set as FORGE, MOD_PLATFORM as AUTO_CURSEFORGE and VERSION as 1.20.1 so it matches with all the mods.
I tried changing my Java version to java 17 because some forge mods seem to have problems with the newer Java versions but it doesn't work either. Also I checked it's all mods, not modpacks.

Lastly I tried doing it with an listing file but no success. I'm completely lost even though I read all the relevant documentation.

**Reply by linthalo77** [reply]
Hey, fixed the issue by omitting the MOD_PLATFORM variable altogether. I probably misunderstood that variable and still don't entirely understand why CurseForge isn't the mod platform? But it works so everythings fine.

Thanks for the help!

**Reply by itzg** [reply]
Glad you got it figured out in the end. 

I'm now realizing I should have called the variable MODPACK_PLATFORM since TYPE is what dictates what type of mods can be loaded. Naming things is hard 😀

**Reply by HAMMERGAMES01** [reply]
> Hey, fixed the issue by omitting the MOD_PLATFORM variable altogether. I probably misunderstood that variable and still don't entirely understand why CurseForge isn't the mod platform? But it works so everythings fine.
> 
> Thanks for the help!

Hello I have the same problem as you can you send me your updated file because I'm not understanding where I'm going wrong

Thanks for the help!

**Reply by linthalo77** [reply]
Hey, unfortunately I have no access to my server rn but if I remember correctly you set your TYPE=FORGE, set your preferred version and delete MOD_PLATFORM alltogether. Then you can list CURSEFORGE_FILES.
